tests/sentry/models/test_lastappconnectbuildscheck.py

```diff
@@ -0,0 +1,156 @@
+import uuid
+from datetime import datetime
+
+import pytest
+from django.utils import timezone
+
+from sentry.lang.native import appconnect
+from sentry.models import LatestAppConnectBuildsCheck
+
+
+class TestCreateOrUpdateBuildCheck:
+    @pytest.fixture
+    def config(self):
+        return appconnect.AppStoreConnectConfig(
+            type="appStoreConnect",
+            id=uuid.uuid4().hex,
+            name="Apple Honk Store Connect",
+            appconnectIssuer="abc123" * 6,
+            appconnectKey="abc123key",
+            appconnectPrivateKey="----BEGIN PRIVATE KEY---- honkhonk",
+            itunesUser="me@example.com",
+            itunesPassword="secrethonk",
+            itunesSession="THE-COOKIE",
+            itunesCreated=datetime.utcnow(),
+            appName="honk beep",
+            appId="123",
+            bundleId="com.honk.app",
+            orgPublicId="71105f98-7743-4844-ab70-2c901e2ea13d",
+            orgName="Honk Com",
+        )
+
+    @pytest.mark.django_db
+    def test_add_new_check_date(self, default_project, config):
+        with pytest.raises(LatestAppConnectBuildsCheck.DoesNotExist):
+            LatestAppConnectBuildsCheck.objects.get(project=default_project, source_id=config.id)
+
+        now = timezone.now()
+        LatestAppConnectBuildsCheck.objects.create_or_update(
+            project=default_project, source_id=config.id, values={"last_checked": now}
+        )
+
+        assert LatestAppConnectBuildsCheck.objects.count() == 1
+
+        try:
+            check_entry = LatestAppConnectBuildsCheck.objects.get(
+                project=default_project, source_id=config.id
+            )
+
+        except LatestAppConnectBuildsCheck.DoesNotExist:
+            raise AssertionError("Build check entry disappeared after update")
```

Comment:
you could not catch the exception and let it go out directly, but i guess this works too :)

Reply:
oh, but if you really feel like doing this the typical way is to call `pytest.fail("Build ...")`

Reply:
done 👍